src/hooks/queries/useEvmFees.ts

```diff
@@ -0,0 +1,68 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { reactQueries } from 'react-queries'
+import { selectEvmFees } from 'react-queries/selectors'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import {
+  assertGetEvmChainAdapter,
+  getFeesWithWallet,
+  type GetFeesWithWalletArgs,
+  isGetFeesWithWalletArgs,
+} from 'lib/utils/evm'
+import {
+  selectFeeAssetByChainId,
+  selectMarketDataByAssetIdUserCurrency,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type UseEvmFeesProps = Omit<
+  GetFeesWithWalletArgs,
+  'wallet' | 'adapter' | 'accountNumber' | 'data'
+> & {
+  accountNumber: number | undefined
+  data: string | undefined
+  chainId: ChainId
+  enabled?: boolean
+  staleTime?: number
+  refetchInterval?: number | false | undefined
+  refetchIntervalInBackground?: boolean
+}
+
+export const useEvmFees = (props: UseEvmFeesProps) => {
+  const wallet = useWallet().state.wallet
+
+  const {
+    enabled = true,
+    staleTime,
+    refetchInterval,
+    refetchIntervalInBackground,
+    ...input
+  } = props
```

Comment:
a new `input` is created every render here, can avoid it using a pattern similar to `stackProps` memoization in `src/components/Layout/Header/NavBar/NavBar.tsx`

Reply:
@woodenfurniture added for paranoia in 393010fdc9 - this one is a similar occurence of technically, we do need memoization here, in reality, given this small component doesn't do much and just returns a react-query (and consumes 2 selectors), we probably don't, but doesn't hurt to I guess!